internals/secrethub/secret_reader.go

```diff
@@ -3,19 +3,39 @@ package secrethub
 import "github.com/secrethub/secrethub-go/pkg/secrethub"
 
 type secretReader struct {
-	client secrethub.Client
+	_client   secrethub.Client
+	newClient newClientFunc
 }
 
 // newSecretReader wraps a client to implement tpl.SecretReader.
-func newSecretReader(client secrethub.Client) *secretReader {
-	return &secretReader{client: client}
+func newSecretReader(newClient newClientFunc) *secretReader {
+	return &secretReader{
+		newClient: newClient,
+	}
 }
 
 // ReadSecret reads the secret using the provided client.
 func (sr secretReader) ReadSecret(path string) (string, error) {
-	secret, err := sr.client.Secrets().Versions().GetWithData(path)
+	client, err := sr.client()
+	if err != nil {
+		return "", err
+	}
+
+	secret, err := client.Secrets().Versions().GetWithData(path)
 	if err != nil {
 		return "", err
 	}
+
 	return string(secret.Data), nil
 }
+
+func (sr secretReader) client() (secrethub.Client, error) {
+	if sr._client == nil {
```

Comment:
Should this functionality reside here or in a higher layer?

Reply:
I've moved it to the client factory. That way, the client used in the secret reader of the run command is also used when fetching secrets for envvars supplied via a flag.